spec/rubocop/cop/rspec/mock_not_stub_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::MockNotStub do
+  subject(:cop) { described_class.new }
+
+  it 'flags stubbed message expectation' do
+    expect_offense(<<-RUBY)
+      expect(foo).to receive(:bar).and_return('hello world')
+                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^ Do not stub your mock.
+    RUBY
+  end
+
+  it 'flags stubbed message expectation with a block' do
+    expect_offense(<<-RUBY)
+      expect(foo).to receive(:bar) { 'hello world' }
```

Comment:
what if the block checks the yielded argument - quite common on exception matching (even though we do not recommend having expectation inside that block)

Reply:
To my best knowledge, `with` supports argument matchers. If it's not possible to use `with`, `and_wrap_original` should be used (only to check arguments and pass to the original).